src/worker.js

```diff
@@ -253,4 +270,7 @@ worker.emit = function (payload) {
 if (typeof exports !== 'undefined') {
   exports.add = worker.register;
   exports.emit = worker.emit;
+  exports.addAbortListener = function(listener) {
+    worker.onAbort = listener;
```

Comment:
I think this makes sense, better to keep with standard listener implementation.

Reply:
At first I was thinking about using `Promise.all`. It is neat though to resolve them one by one in order. I think there is no need to extract and pass `first` to the `reduce` method: if you omit the inital value, it will automatically pick the first element of the array as initial value. And I think you can write the body of the `reduce` method more compact, like `.reduce((prev, next) => prev.then(next))`. 

If one of the abort listeners throws an exepction, I think that we should fall back on running `_exit();`, what do you think?

Reply:
I tried the reduce and I think its easier to do something like this

```js
    const promises = worker.abortListeners.map((listener) => listener());
    Promise.race([
          Promise.allSettled(promises).then(_abort),
          () => {
            return new Promise((resolve) => setTimeout(resolve, 300))  
          }
    ]).then(_abort, _exit);
```

So we just execute the listeners and wrapped promises with `Promise.allSettled` and just wait for all promises to fulfill before calling `_abort` or `_exit` respectfully,  I don't think sequential chaining is worth it as it will make handling the timeout a bit tricky.

Reply:
Agree, I think multiple listeners is a very small edge case already, let's keep it simple and resolve them in parallel 👍 

Your code example looks good indeed, with the `Promise.race`. And then the 300 ms can become a configurable option.

Reply:
I ended up being able to use the `all` implementation within our `Promise` implementation. I refactored to use it as adding new Promise methods within this pr felt out of scope and turned out not to be necessary.